Source/Parallelization/WarpXComm.cpp

```diff
@@ -610,15 +619,25 @@ WarpX::FillBoundaryB (int lev, PatchType patch_type, IntVect ng)
 {
     if (patch_type == PatchType::fine)
     {
-        if (do_pml && pml[lev]->ok())
+        if (do_pml)
         {
-            pml[lev]->ExchangeB(patch_type,
-                            { Bfield_fp[lev][0].get(),
-                              Bfield_fp[lev][1].get(),
-                              Bfield_fp[lev][2].get() },
-                              do_pml_in_domain);
-        pml[lev]->FillBoundaryB(patch_type);
+            if (pml[lev] && pml[lev]->ok())
+            {
+                pml[lev]->ExchangeB(patch_type,
+                                { Bfield_fp[lev][0].get(),
+                                  Bfield_fp[lev][1].get(),
+                                  Bfield_fp[lev][2].get() },
+                                  do_pml_in_domain);
+            pml[lev]->FillBoundaryB(patch_type);
+            }
+#ifdef WARPX_DIM_RZ
+            if (pml_rz[lev])
+            {
+                pml_rz[lev]->FillBoundaryB(patch_type);
```

Comment:
Why do we not need to exchange data between the valid domain and the PML, as we do in Cartesian geometry (see call to `ExchangeB` function above)?